csrc/evaluator_common.cpp

```diff
@@ -103,6 +103,8 @@ std::vector<Val*> collectRuntimeUsedValues(Fusion* fusion) {
         ret.push_back(id->expandedExtent());
       }
     }
+    Val* metadata = fusion->metadataOf(tv);
+    ret.push_back(metadata);
```

Comment:
We were not binding metadata to `PrecomputedValues`, which I think is unexpected. This fixes that (see change to `TensorMetaData` test).

Reply:
On second thought, I'm splitting this into another small PR, with the accompanying test from #2714.

Reply:
See #2812